source/widgets/browsing/Pagination.tsx

```diff
@@ -28,7 +47,21 @@ const Pagination = (props: IPaginationProps) => {
         onClick={() => onChangePage(currentPage - 1)}
       />
       <div className={styles.pageInfo}>
-        <span className={styles.pageNumber}>{currentPage}</span>
+        <span className={styles.pageNumber}>
+          <span
+            contentEditable={true}
```

Comment:
Contenteditable is not recommended with React and this actually has a bug: simply select the current value and replace it with another number (like 20) -> you get "2020" in the editable area (i guess that's because react doesn't know about the edited content, and adds the `currentPage` value.

Please refactor to a normal input component 🙏

Reply:
We can't use input component for this auto width because it cant be styled in the same way like on the design.

With contentEditable i know there are some performance issues, but i tried to catch the onKeyPress event, change the page and stop the rest of event execution with                 e.preventDefault();
e.stopPropagation();

Another thing i am doing is rechecking when the componentDidUpdate and if the property received for currentPage is different then the DOM content of the span which holds the current page, it will just try to force it to the new props and that solved the issue.

In this way we don't have any issue with duplication and current page is always loaded well.

Reply:
Please disregard my last comment. @DominikGuzei